### PR TITLE
Restore the ability to re initialise the calendar programmatically

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue #15251 

### Summary of Changes
Minor changes to the calendar script that allow reinitialising the calendar

### Testing Instructions
Apply this patch
Go to Administrator and edit an existing article.
Open the browser inspector and inspect the **button** of the Start Publishing field.
Then change data-show-others="1"  to "0"
Then run this in the browser's console:
```js
JoomlaCalendar.init(document.querySelector('.field-calendar'));
```
the cal should reinitialise, to confirm open it again you should see the table without the days for prev/next month

Alternative test method (only js, also a hint how to do this client side):
Apply this patch
Go to Administrator and edit an existing article.
Open the browser console and copy/paste the following commands
```js
document.getElementById('jform_publish_up_btn').setAttribute('data-show-others', 0);
JoomlaCalendar.init(document.querySelector('.field-calendar'));
```
the cal should reinitialise, to confirm open it again you should see the table without the days for prev/next month

### Documentation Changes Required

### CODE REVIEW

**Comment by zero-24** [comment]
The drone error has nothing to do with this change I'm going to send a new PR for this. Thanks @dgt41

**Comment by chmst** [comment]
I've tested this patch following the instruction above. And it works as decribed. 
But as much as I see, it does not resolve the issue. The calendars in our programs are not reinizialised. So please let me know what we have to change in our programs.

**Comment by chmst** [comment]
After changing the code, the calendar is initialised. But ina format which is not expected.
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15487">issues.joomla.org/tracker/joomla-cms/15487</a>.</sub>

**Comment by dgt41** [comment]
@chmst let's say that ajax success returns a date, all you have to do is set the appropriate attribute and initialise the calendar again. A dummy code (using jQuery):
```js
// For value we target the input
var newDate = "2010-01-01 00:00:00"; // This comes from AJAX request

// For calendar attributes we target the input id + `_btn`
jQuery('#'jform_publish_up_btn'). data('show-others', 0);

// The field node
$field = jQuery('#jform_publish_up');
// The calendar node
$calElement = $field.parent('.field-calendar');

// Resetting attributes
$field.val(newDate);
$field.data('alt-value', newDate);

// Reinitialise
JoomlaCalendar.init($calElement[0]); // $calElement[0] will return the vanilla node!!!
```

Is this any helpful? If something still unclear let me know

**Comment by richard67** [comment]
Since the original issue #15251 had the release blocker label, shouldn't this PR also have that?

**Comment by dgt41** [comment]
@chmst what properties/values are you resetting with your code?

**Comment by bembelimen** [comment]
The test case is still very easy:

Ajax request => call: https://github.com/joomla/joomla-cms/issues/15251#issuecomment-293578934

In 3.6 the calender works, in 3.7 not...

**Comment by dgt41** [comment]
@bembelimen in 3.7 this code is the equivalent https://gist.github.com/dgt41/88a1a5fda0e88ff07e0723d85118fdba

**Comment by bembelimen** [comment]
It's still not B/C, wrong date format etc.

**Comment by rdeutz** [comment]
To be crystal clear here, I am happy if we find a way to handle the problem but I will not delay the release for this problem.

**Comment by chmst** [comment]
You are the boss ;) It works for me after I've adapted my programs for setting attributes in JHtml and replacing the calendar setup in js. 
So for me the patch is successfull but the issue (b/c break) is a fact.

**Comment by rdeutz** [comment]
@chmst good that you found a way to solve the issue. :-)

**Comment by zero-24** [comment]
I'm going to create a FAQ page based on the infos above when i'm back at home. Thanks for the work here!

**Comment by chmst** [comment]
Thanks to all - I am saved :)

**Comment by bembelimen** [comment]
So will the original issue reopened again? Because the b/c is still not fixed...

**Comment by dgt41** [comment]
@bembelimen don't worry, I'm creating a PR that will cover the B/C break here, gimme 5 mins

**Comment by dgt41** [comment]
@bembelimen please test #15489

File: media/system/js/fields/calendar.js [hunk 1]
```diff
@@ -1113,16 +1117,22 @@
...
+		var elements, i;
+
+		elements = document.querySelectorAll(".field-calendar");
+
+		for (i = 0; i < elements.length; i++) {
+			JoomlaCalendar.init(elements[i]);
+		}
 
 		window.jQuery && jQuery(document).on("subform-row-add", function (event, row) {
 			JoomlaCalendar.init(".field-calendar", row);
```
Comment (Fedik) [inline media/system/js/fields/calendar.js]: @dgt41 you missed something 😉

**Pull request merged**